BrWizardNavigation.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <div class="row justify-center width-100 q-mt-lg">
+    <back-button
+      v-if="currentStepIndex > 0"
+      class="q-mr-md"
+      :disabled="blockBack"
+      @back="onBack()" />
+    <next-button
+      v-if="currentStepIndex < totalSteps - 1"
+      :class="{'q-ml-md': currentStepIndex > 0}"
+      :disabled="blockNext"
+      @next="onNext()" />
+    <finish-button
+      v-if="currentStepIndex === totalSteps - 1"
+      class="q-ml-md"
+      :disabled="blockFinish"
+      @finish="onFinish()" />
+  </div>
+</template>
+
+<script>
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+import NextButton from './NextButton.vue';
+import BackButton from './BackButton.vue';
+import FinishButton from './FinishButton.vue';
+
+export default {
+  name: 'BrWizardNavigation',
+  components: {NextButton, BackButton, FinishButton},
+  props: {
+    blockBack: {
+      type: Boolean,
+      required: true
+    },
+    blockFinish: {
+      type: Boolean,
+      required: true
+    },
+    blockNext: {
+      type: Boolean,
+      required: true
+    },
+    currentStepIndex: {
+      type: Number,
+      required: true
+    },
+    totalSteps: {
+      type: Number,
+      required: true
+    },
+  },
+  methods: {
+    async onNext() {
+      if(!await this._emitCancelableEvent('next', {
```

Comment:
I don't think I like putting an emitter like this in a conditional.  Feels different to me from putting a regular function in there because the emitter is *designed* to produce side effects right?  Is this a common pattern in this framework?  I would prefer to see `const cancelled = await ...` and then the conditional.